### ISSUE TITLE
Documentation - getting started

### ISSUE THREAD

**Issue by jbulow** [gtfierro/hoddb#5]
There are no instructions on how to start or use the server.

The current Makefile does not even build a hod executable nut a "log" executable. Is "log" the new name of the server? When starting "log" it does not seem to follow the port specification in the configuration file. The hodconfig.yml specifies port 47808 but the "log" server listens to 47809. Also, it does only listen to localhost and not the address specified in the config file.

**Comment by gtfierro** [comment]
You're right! There's a lot of clean up that still needs to happen, and I hope to get to it soon

**Comment by jbulow** [comment]
Can I help or assist you in this work?

**Comment by HackerBaloo** [comment]
I managed to build it with just a standard go setup:
https://golang.org/doc/install
and then:
go build
in the working copy
and then run with:
./hoddb

**Comment by HackerBaloo** [comment]
But on the same subject, if I want to test it as a sparql endpoint, do I need to configure anything?
It's serving on localhost:47809
So the sparql endpoint is? http://localhost:47809 or http://localhost:47809/query or http://localhost:47809/sparql
and should I enable http in hodconfig.yaml?

**Comment by gtfierro** [comment]
I've been travelling and preparing for an exam this past month, so I'll be able to jump in more readily in a week or so, but here's some quick responses and directions.

The API is GRPC (defined [here](https://github.com/gtfierro/hoddb/blob/master/proto/log.proto)). The [frontend code](https://github.com/gtfierro/hoddb/blob/master/hod/frontend.go) has 47808 as the GRPC port and 47809 is a HTTP gateway to the GRPC service. HodDB does not currently support the official SPARQL protocol (or even the full SPARQL language! All of HodDB's queries need a `;` at the end because when I wrote the parser I forgot there was no `;` and haven't had the time to fix it). This is not fundamental, but I would like a GRPC-based interface because it is easier to generate clients for.

There are some fixes and experimental features in the [p2p branch](https://github.com/gtfierro/hoddb/blob/p2p) which I hope to start bringing over into the master branch, so if something breaks/doesn't work it *might* be fixed in the other branch.
